File: Java/DeepSpace/src/deepspace/ShieldBooster.java
package deepspace;

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 *
 * @author victor
 */
class ShieldBooster {
    private String name;
    private float boost;
    private int uses;

    ShieldBooster(String name, float boost, int uses) {
        this.name = name;
        this.boost = boost;
        this.uses = uses;
    }
    
    ShieldBooster(ShieldBooster s) {
        name = s.name;
        boost = s.boost;
        uses = s.uses;
    }
    
    ShieldToUI getUIversion() {
        return new ShieldToUI(this);
    }

    public float getBoost() {
        return boost;
    }

    public int getUses() {
        return uses;
    }
    
    float useIt() {
        if(uses > 0) {
            uses--;
            return boost;
        } else
            return 1.0f;            
    }

    @Override
    public String toString() {
        return "ShieldBooster{" + "name=" + name + ", boost=" + boost + ", uses=" + uses + '}';
    }
    
    
    
    
}

File: Ruby/Deepspace/lib/ShieldBooster.rb
#encoding:utf-8
module Deepspace
class ShieldBooster
  attr_reader :name, :boost, :uses
  
  def initialize(name, boost, uses)
    @name = name
    @boost = boost
    @uses = uses    
  end
  
  def self.newCopy(copy)
    ShieldBooster.new(copy.name, copy.boost,copy.uses)
  end
  
  def useIt
    if @uses > 0
      @uses -= 1
      @boost
    else
      1.0
    end
  end
  
  def to_s
    "\nNombre: #{@name} Boost: #{@boost} Uses: #{@uses}"
  end
  
  def getUIversion
    ShieldToUI.new(self)
  end
  
  def to_s
       return "\nName = #{@name} Boost = #{@boost} Uses = #{@uses}"
  end
  
end
end
